### PR TITLE
[SFLOPPY] Move and rename floppy_new to class/sfloppy

### PR DESCRIPTION
**Pull request by SergeGautherie**
Supersedes #3509.

JIRA issue: [CORE-17411](https://jira.reactos.org/browse/CORE-17411)

### CODE REVIEW

File: boot/bootdata/hivesys.inf [hunk 1]
```diff
@@ -1394,7 +1394,7 @@ HKLM,"SYSTEM\CurrentControlSet\Control\ServiceGroupOrder","List",0x00010000, \
  "System Bus Extender", \
  "SCSI Miniport", \
  "Port", \
- "Primary Disk", \
+ "Primary disk", \
```
Comment (ColinFinck) [inline boot/bootdata/hivesys.inf:1397]: This is just wrong, and unrelated.
Why are you mixing this into a PR about the sfloppy driver??

**Review by ColinFinck — CHANGES_REQUESTED** [review]

File: boot/bootdata/hivesys.inf (hunk 1 above)
Comment (SergeGautherie) [reply boot/bootdata/hivesys.inf]: Indeed, sorry, reverted that one.
Because floppy_new has that mismatch...

File: boot/bootdata/hivesys.inf [hunk 2]
```diff
@@ -1773,7 +1773,7 @@ HKLM,"SYSTEM\CurrentControlSet\Services\usbuhci","Type",0x00010001,0x00000001
 
 ; USB storage driver
 HKLM,"SYSTEM\CurrentControlSet\Services\usbstor","ErrorControl",0x00010001,0x00000001
-HKLM,"SYSTEM\CurrentControlSet\Services\usbstor","Group",0x00000000,"Primary Disk"
+HKLM,"SYSTEM\CurrentControlSet\Services\usbstor","Group",0x00000000,"Primary disk"
```
Comment (HBelusca) [inline boot/bootdata/hivesys.inf:1776]: Please revert.

File: drivers/storage/floppy/floppy/floppy_reg.inf [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ; Floppy driver
 [AddReg]
 HKLM,"SYSTEM\CurrentControlSet\Services\Floppy","ErrorControl",0x00010001,0x00000000
-HKLM,"SYSTEM\CurrentControlSet\Services\Floppy","Group",0x00000000,"Primary Disk"
+HKLM,"SYSTEM\CurrentControlSet\Services\Floppy","Group",0x00000000,"Primary disk"
```
Comment (HBelusca) [inline drivers/storage/floppy/floppy/floppy_reg.inf:4]: Please revert.

File: drivers/storage/class/sfloppy/sfloppy_reg.inf [hunk 4]
```diff
@@ -0,0 +1,6 @@
+; Microsoft Super Floppy (sfloppy) Storage Class Driver
+[AddReg]
+HKLM,"SYSTEM\CurrentControlSet\Services\Sfloppy","ErrorControl",0x00010001,0x00000000
+HKLM,"SYSTEM\CurrentControlSet\Services\Sfloppy","Group",0x00000000,"Primary disk"
```
Comment (HBelusca) [inline drivers/storage/class/sfloppy/sfloppy_reg.inf:4]:
```suggestion
HKLM,"SYSTEM\CurrentControlSet\Services\Sfloppy","Group",0x00000000,"Primary Disk"
```

**Review by HBelusca — CHANGES_REQUESTED** [review]

**Comment by Extravert-ir** [comment]
sfloppy is a class (classpnp) driver, so it should reside in a `drivers/class` folder

File: boot/bootdata/hivesys.inf (hunk 1 above)
Comment (Extravert-ir) [reply boot/bootdata/hivesys.inf]: The right writing is `Primary Disk`. Please fill a separate PR with that

Comment (SergeGautherie) [reply boot/bootdata/hivesys.inf]: Moved to #3516...

**Comment by SergeGautherie** [comment]
> `drivers/class` folder

Moved.

**Comment by SergeGautherie** [comment]
Added 'Tag' key.

**Comment by HBelusca** [comment]
The definitive correct folder is `drivers/storage/class`. Indeed:
1. It's a class driver for the storage stack, not for whatever generic driver.
2. All the other storage class drivers we have (some of which, as you know, coming from the MS samples) are already in the existing `drivers/storage/class`.

**Comment by Extravert-ir** [comment]
It is `drivers/storage/class` ofc, I mistyped
BTW, as we're here, it makes sense to write an inf file for the driver (in order for it to be complete). It should take `GenSFloppy`, `USBSTOR\GenSFloppy` and some LS-120/SuperDrive IDs

**Comment by HBelusca** [comment]
For completeness, at least on Windows 10 the sfloppy INF data stuff that @Extravert-ir mentions, is present inside a `flpydisk.inf` file.
We happen to have this file already:
https://git.reactos.org/?p=reactos.git;a=blob;f=media/inf/flpydisk.inf;hb=0c34c067c461d00c290e76e8066b226e47eeec43

**Comment by Extravert-ir** [comment]
I want it to be a separate file, otherwise we will have to always ship both flpydisk and sfloppy

**Comment by SergeGautherie** [comment]
> `drivers/storage/class` ofc

Hum, isn't that what I had submitted initially? Or do you mean you do not want the `/sfloppy` sub-directory?

> write an inf file for the driver

I would prefer to only move and update existing files,
and leave further dev and possibly enabling the driver to a follow-up PR.

**Comment by ColinFinck** [comment]
> I want it to be a separate file, otherwise we will have to always ship both flpydisk and sfloppy together

Do we know what "flpydisk" does exactly, and why NT4's "floppy" driver was split into "fdc" and "flpydisk"?

I haven't checked the details yet, but I could imagine that "flpydisk" provides common code for the different interfaces that support floppies, like "fdc" or "sfloppy".
This is pure speculation though. Some enlightenment would be very helpful here.

**Comment by SergeGautherie** [comment]
> I could imagine that "flpydisk" provides common code

For what it's worth, that looks likely:

http://dll-repair.com/flpydisk-sys/
'flpydisk.sys, also written as Floppy Driver, is used to support fdc, fdc.sys and sfloppy, sfloppy.sys [...]'

http://www.delphigroups.info/2/7c/502169.html
'We did download the Win2k DDK and found no floppy driver but 1. fdc.sys + 2. flpydisk.sys instead. [...]'

**Pull request closed without merging**